Assets/HoloToolkit-Examples/GazeRuler/Scripts/LineManager.cs

```diff
@@ -0,0 +1,137 @@
+﻿using UnityEngine;
+using System.Collections;
+using System;
+using HoloToolkit.Unity;
+using System.Collections.Generic;
+
+/// <summary>
+/// mananger all lines in the scene
+/// </summary>
+public class LineManager : Singleton<LineManager>, IGeometry
+{
+    // save all lines in scene
+    private Stack<Line> Lines = new Stack<Line>();
+
+
+    private Point lastPoint;
+
+
+    // place point and lines
+    public void AddPoint(GameObject LinePrefab, GameObject PointPrefab, GameObject TextPrefab)
+    {
+
+        var hitPoint = GazeManager.Instance.HitInfo.point;
+
+
+        var point = (GameObject)Instantiate(PointPrefab, hitPoint, Quaternion.identity);
+        if (lastPoint != null && lastPoint.IsStart)
+        {
+            var centerPos = (lastPoint.Position + hitPoint) * 0.5f;
+
+            var directionFromCamera = centerPos - Camera.main.transform.position;
+
+            var distanceA = Vector3.Distance(lastPoint.Position, Camera.main.transform.position);
+            var distanceB = Vector3.Distance(hitPoint, Camera.main.transform.position);
+
+            Debug.Log("A: " + distanceA + ",B: " + distanceB);
+            Vector3 direction;
+            if (distanceB > distanceA || (distanceA > distanceB && distanceA - distanceB < 0.1))
+                direction = hitPoint - lastPoint.Position;
+            else
+                direction = lastPoint.Position - hitPoint;
+
+            var distance = Vector3.Distance(lastPoint.Position, hitPoint);
+            var line = (GameObject)Instantiate(LinePrefab, centerPos, Quaternion.LookRotation(direction));
+            line.transform.localScale = new Vector3(distance, 0.005f, 0.005f);
+            line.transform.Rotate(Vector3.down, 90f);
+
+            var normalV = Vector3.Cross(direction, directionFromCamera);
+            var normalF = Vector3.Cross(direction, normalV) * -1;
+
+
+            var tip = (GameObject)Instantiate(TextPrefab, centerPos, Quaternion.LookRotation(normalF));
+
+            //unit is meter
+            tip.transform.Translate(Vector3.up * 0.05f);
+            tip.GetComponent<TextMesh>().text = distance + "m";
```

Comment:
I suspect this is what showed up as the character in your video :) You locale might be set different :)